node/src/blockchain/blockchain_interface/blockchain_interface_web3/lower_level_interface_web3.rs

```diff
@@ -612,6 +658,74 @@ mod tests {
         assert_eq!(tx_receipt.status, TxStatus::Pending);
     }
 
+    #[test]
+    fn tx_status_display_works() {
+        // Test Failed
+        assert_eq!(TxStatus::Failed.to_string(), "Failed");
+
+        // Test Pending
+        assert_eq!(TxStatus::Pending.to_string(), "Pending");
+
+        // Test Succeeded
+        let block_number = U64::from(12345);
+        let block_hash = H256::from_low_u64_be(0xabcdef);
+        let succeeded = TxStatus::Succeeded(TransactionBlock {
+            block_hash,
+            block_number,
+        });
+        assert_eq!(
+            succeeded.to_string(),
+            format!("Succeeded({},0x{:x})", block_number, block_hash)
+        );
+    }
+
+    #[test]
+    fn tx_status_from_str_works() {
+        // Test Pending
+        assert_eq!(TxStatus::from_str("Pending"), Ok(TxStatus::Pending));
+
+        // Test Failed
+        assert_eq!(TxStatus::from_str("Failed"), Ok(TxStatus::Failed));
+
+        // Test Succeeded with valid input
+        let block_number = 123456789;
+        let block_hash = H256::from_low_u64_be(0xabcdef);
+        let input = format!("Succeeded({},0x{:x})", block_number, block_hash);
+        assert_eq!(
+            TxStatus::from_str(&input),
+            Ok(TxStatus::Succeeded(TransactionBlock {
+                block_hash,
+                block_number: U64::from(block_number),
+            }))
+        );
+
+        // Test Succeeded with invalid format
+        assert_eq!(
+            TxStatus::from_str("Succeeded(123)"),
+            Err("Invalid Succeeded format".to_string())
+        );
+
+        // Test Succeeded with invalid block number
+        assert_eq!(
+            TxStatus::from_str(
+                "Succeeded(abc,0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef)"
+            ),
+            Err("Invalid block number".to_string())
+        );
+
+        // Test Succeeded with invalid block hash
+        assert_eq!(
+            TxStatus::from_str("Succeeded(123,0xinvalidhash)"),
+            Err("Invalid block hash".to_string())
+        );
+
+        // Test unknown status
+        assert_eq!(
+            TxStatus::from_str("InProgress"),
+            Err("Unknown status: InProgress".to_string())
+        );
+    }
```

Comment:
Since these are basically a set of alone-assertions I'd suggest to delete all the blank lines, I think it would work as well as this and be more concise.

Please consider that at the both of these tests above. 

You know what... do as you please to. I don't have a strong opinion on this. You can easily leave it this way. Cheers.